tests/unit/core/server/testServer.js

```diff
@@ -23,32 +36,47 @@ describe('core/server/base', () => {
     global.webpackIsomorphicTools = {
       assets: () => fakeAssets,
     };
+    mockWindow = sinon.mock(global.window);
   });
 
   afterEach(() => {
-    Helmet.canUseDOM = _helmentCanUseDOM;
+    Helmet.canUseDOM = _helmetCanUseDOM;
     delete global.webpackIsomorphicTools;
   });
 
-  function testClient({ stubRoutes = defaultStubRoutes } = {}) {
-    function createStoreAndSagas() {
-      const sagaMiddleware = createSagaMiddleware();
-      return {
-        store: createStore(
-          combineReducers({ reduxAsyncConnect }),
-          // Do not define an initial state.
-          undefined,
-          applyMiddleware(sagaMiddleware),
-        ),
-        sagaMiddleware,
-      };
+  function createStoreAndSagas({
+    reducers = { reduxAsyncConnect },
+  } = {}) {
+    const sagaMiddleware = createSagaMiddleware();
+    const store = createStore(
+      combineReducers(reducers),
+      // Do not define an initial state.
+      undefined,
+      applyMiddleware(sagaMiddleware),
+    );
+
+    return { store, sagaMiddleware };
+  }
+
+  function testClient({
+    stubRoutes = defaultStubRoutes,
+    store = null,
+    sagaMiddleware = null,
+    appSagas = null,
+  } = {}) {
+    function _createStoreAndSagas() {
+      if (store === null) {
+        return createStoreAndSagas();
+      }
+
+      return { store, sagaMiddleware };
     }
 
     // eslint-disable-next-line no-empty-function
     function* fakeSaga() {}
 
-    const app = baseServer(stubRoutes, createStoreAndSagas, {
-      appSagas: fakeSaga,
+    const app = baseServer(stubRoutes, _createStoreAndSagas, {
+      appSagas: appSagas || fakeSaga,
```

Comment:
Can you also try to add test coverage for the `disableSSR=true` case? I think you could do it with this patch:

```js
diff --git a/src/core/server/base.js b/src/core/server/base.js
index e2f727a9..5927a744 100644
--- a/src/core/server/base.js
+++ b/src/core/server/base.js
@@ -3,7 +3,7 @@ import path from 'path';
 
 import 'babel-polyfill';
 import { oneLine } from 'common-tags';
-import config from 'config';
+import defaultConfig from 'config';
 import Express from 'express';
 import helmet from 'helmet';
 import Raven from 'raven';
@@ -116,7 +116,10 @@ function hydrateOnClient({ res, props = {}, pageProps }) {
     .end();
 }
 
-function baseServer(routes, createStore, { appSagas, appInstanceName = appName } = {}) {
+function baseServer(routes, createStore, {
+  appSagas, appInstanceName = appName,
+  config = defaultConfig,
+} = {}) {
   const app = new Express();
   app.disable('x-powered-by');
 
```

Reply:
This works and I see the coverage in the report, but I haven't found a good assertion yet.